### PR TITLE
Add logical type support to Json converter

### PR DESCRIPTION
**Pull request by ikbalkaya**
Adds [Avro logical type](https://avro.apache.org/docs/1.10.2/spec.html#Logical+Types) support when converting schematic data into JSON
Please note that when converting from or to Avro schema, I got inspiration from [this particular class ](https://github.com/confluentinc/schema-registry/blob/4095b46188f98fcc1e12e8b8b3fc1ee0c2c1e6d4/client/src/main/java/io/confluent/kafka/schemaregistry/avro/AvroSchemaUtils.java#L89). 
For date types there were different resulting types from my tests and types defined under `org.apache.avro.data`. As a result some logical type conversions will expect more than one types (for example both `Instant` and a `Date` is possible for `timestamp-millis`). The source of the truth for this is test files that has Avro logical types in them and test for this work will confirm the output JSON is exactly the same as the input one. 

This also tidies up docker-compose file by adding container names. Also un-skips tests when building the project.

Fixes https://github.com/ably/kafka-connect-ably/issues/97

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -251,6 +251,12 @@
       <scope>test</scope>
     </dependency>
 
+    <!-- https://mvnrepository.com/artifact/io.confluent/kafka-connect-avro-data -->
+    <dependency>
+      <groupId>io.confluent</groupId>
+      <artifactId>kafka-connect-avro-data</artifactId>
```
Comment (jaley) [inline pom.xml:257]: Should this dependency have `<scope>test</scope>`? I don't see (and didn't expect) any runtime Avro dependencies added for this change?

File: src/main/java/com/ably/kafka/connect/utils/LogicalTypeConversions.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+import java.math.BigDecimal;
+import java.time.Instant;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.ZoneOffset;
+import java.util.Date;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+class LogicalTypeConversions{
```
Comment (jaley) [inline src/main/java/com/ably/kafka/connect/utils/LogicalTypeConversions.java:14]: + newline?

File: src/main/java/com/ably/kafka/connect/utils/StructToJsonConverter.java [hunk 3]
```diff
@@ -79,13 +81,16 @@ private static Map<String, Object> structJsonMap(final Struct struct) {
...
                     structMap.put(field.name(), struct.getBoolean(field.name()));
                     break;
-                case STRING:
-                    structMap.put(field.name(), struct.getString(field.name()));
-                    break;
                 case BYTES:
-                    throw new ConnectException("Bytes are currently not supported for conversion to JSON.");
+                    final Object logicalValue = LogicalTypeConversions.tryGetLogicalValue(schemaType, struct.get(field.name()));
+                    if (!(logicalValue instanceof BigDecimal)) {
+                        throw new ConnectException("Bytes are currently not supported for conversion to JSON.");
```
Comment (jaley) [inline src/main/java/com/ably/kafka/connect/utils/StructToJsonConverter.java:87]: `ConnectionException` doesn't feel like a very descriptive exception type for this error. Do we have nothing more appropriate?

**Review by jaley — APPROVED** [review]
Couple of questions below, but overall looks fine. Thanks!

File: src/main/java/com/ably/kafka/connect/utils/LogicalTypeConversions.java [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+                    LocalDate localDate = date.toInstant().atZone(ZoneOffset.UTC).toLocalDate();
+                    return (int)localDate.toEpochDay();
+                } else if (value instanceof LocalDate) {// date logical type
+                     final LocalDate date = (LocalDate) value;
+                     return (int)date.toEpochDay();
+                 } else if (value instanceof LocalTime) {//time-millis and time-micros
+                     final LocalTime time = (LocalTime) value;
+                     return (int) TimeUnit.NANOSECONDS.toMillis(time.toNanoOfDay());
+                 }
+            case INT64:
```
[the remark this answers is not in the document]
Comment (ikbalkaya) [reply src/main/java/com/ably/kafka/connect/utils/LogicalTypeConversions.java:39]: That's a good spot. This was initially intentional as I wanted to try using the same conversions for all INT types but I decided to separate them to make outputs explicitly ints or longs which I must have forgotten. I made the change here.  d08a2f8f2aed7f64ec833193eb9c3cc768dc3846 .

File: src/main/java/com/ably/kafka/connect/utils/StructToJsonConverter.java (hunk 3 above)
Comment (ikbalkaya) [reply src/main/java/com/ably/kafka/connect/utils/StructToJsonConverter.java:87]: This is the generic `ConnectException` that we use for connector exceptions, but yes I think we could create/use something a bit more descriptive and also an exception that belongs to our package. I'm thinking about something like `DataFormatException` 🤔 WDYT?

File: pom.xml (hunk 1 above)
Comment (ikbalkaya) [reply pom.xml:257]: Yes definitely. I was messing up with this to see if I can use at converter itself but then realized it was not necessary :) Made change ea4c7bc522d4ae47844b26fd07edc6003e8b6ee0

File: src/main/java/com/ably/kafka/connect/utils/LogicalTypeConversions.java [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+import java.util.concurrent.TimeUnit;
+class LogicalTypeConversions{
+    /*
+    *This method will try to get Avro logical type of
+    * @param value and if it exists it will return a value that that has a sensible serialization format.
+    * @param type is the normal Avro 'type where logical types applies ..
+    * Check @link https://avro.apache.org/docs/1.10.2/spec.html#Logical+Types
+    * for the reference on logical types.
+    * */
+     static Object tryGetLogicalValue(final Schema.Type type, Object value) {
```
[the remark this answers is not in the document]
Comment (ikbalkaya) [reply src/main/java/com/ably/kafka/connect/utils/LogicalTypeConversions.java]: fed3ec16379e53c6dc9e56a9fe655334fa9371e1

File: src/main/java/com/ably/kafka/connect/utils/LogicalTypeConversions.java (hunk 2 above)
Comment (ikbalkaya) [reply src/main/java/com/ably/kafka/connect/utils/LogicalTypeConversions.java:14]: 7c2eec99a88f9415e89e45b3f5ec93c097112b42

Comment (ikbalkaya) [reply src/main/java/com/ably/kafka/connect/utils/LogicalTypeConversions.java:14]: 7c2eec99a88f9415e89e45b3f5ec93c097112b42

File: src/main/java/com/ably/kafka/connect/utils/StructToJsonConverter.java (hunk 3 above)
Comment (ikbalkaya) [reply src/main/java/com/ably/kafka/connect/utils/StructToJsonConverter.java]: I went with `UnsupportedJsonTypeException` 86c229d2061c2c189fb4e430591c1086c1f66fde

Comment (jaley) [reply src/main/java/com/ably/kafka/connect/utils/StructToJsonConverter.java]: Ah thanks, I didn't realise `Connect` here is actually the proper noun / brand name `Connect` as opposed to describing a network connection problem! This change seems good to me in any case.

**Review by QuintinWillison — APPROVED** [review]

**Pull request merged**